packages/mui-base/src/TabUnstyled/TabUnstyled.types.ts

```diff
@@ -39,17 +39,8 @@ export interface TabUnstyledSlots {
   root?: React.ElementType;
 }
 
-export type TabUnstyledProps<
-  D extends React.ElementType = TabUnstyledTypeMap['defaultComponent'],
-  P = {},
-> = OverrideProps<TabUnstyledTypeMap<P, D>, D> & {
-  component?: D;
-};
-
-export interface TabUnstyledTypeMap<P = {}, D extends React.ElementType = 'button'> {
-  props: P & TabUnstyledOwnProps;
-  defaultComponent: D;
-}
+export type TabUnstyledProps = TabUnstyledOwnProps &
```

Comment:
We discussed also to make the base components accept generic for the root slot, so that the root props can be spread directly on the component. Do you want to try to implement this? If not I can help out trough out the week and add it in the `Button` PR.

Reply:
I can also look into this today, let you know what progress I've made in an hour or two :)